cvlr-spec/src/lib.rs

```diff
@@ -0,0 +1,71 @@
+//! Specification language for CVL (Certora Verification Language) in Rust.
+//!
+//! This module provides a framework for writing specifications with preconditions
+//! (requires) and postconditions (ensures) that can be used for formal verification.
+//!
+//! # Core Concepts
+//!
+//! ## Boolean Expressions
+//!
+//! The [`CvlrBoolExpr`] trait represents boolean expressions that can be:
+//! - Evaluated to a boolean value
+//! - Asserted (checked for correctness)
+//! - Assumed (taken as preconditions)
+//!
+//! ## Composing Expressions
+//!
+//! Boolean expressions can be composed using:
+//! - [`cvlr_and`] - Logical AND
+//! - [`cvlr_impl`] - Logical implication (A → B)
+//! - [`cvlr_true`] - Constant true expression
+//!
+//! ## State Pairs
+//!
+//! Postconditions use [`eval_with_states`](CvlrBoolExpr::eval_with_states) to evaluate
+//! over both pre-state and post-state contexts, allowing you to express postconditions
+//! that compare states before and after operations.
+//!
+//! ## Specifications
+//!
+//! The [`CvlrSpec`] trait represents a complete specification with:
+//! - Preconditions (requires) - conditions that must hold before an operation
```

Comment:
minor thing but maybe change the description of requires to "conditions that are assumed before an operation".
Having the same description for requires and ensures is weird since they do different things